### PR TITLE
Ajax: Run the PATCH test only in IE8 on TestSwarm

### PR DESCRIPTION
**Pull request by mzgol**
Fixes gh-1994

### CODE REVIEW

File: test/unit/ajax.js [hunk 1]
```diff
@@ -1578,16 +1578,19 @@ module( "ajax", {
...
-		url: "data/echoQuery.php",
-		method: "PATCH",
-		success: function() {
-			ok( true, "success" );
-		},
-		error: function() {
-			ok( false, "error" );
-		}
-	});
+	// BrowserStack PATCH support sometimes breaks so on TestSwarm run the test in IE8 only.
```
Comment (markelog) [inline test/unit/ajax.js]: Hm, so browserstack can't fix that issue huh? Would need to be careful here when we switch from testswarm

Comment (mzgol) [inline test/unit/ajax.js]: Well, I've already exchanged 22 mails with BrowserStack people, the first one at the end of October. This problem disappears and then re-appears again after some time. I'd rather not spend more time on this so it's better to restrict this to IE8.

As for switching from TestSwarm: sure, though if we forget about it then nothing terrible will happen. We'll notice it the first time PATCH fails again. ;) And then we'll fix the exclusion logic.

Comment (markelog) [inline test/unit/ajax.js]: > Well, I've already exchanged 22 mails with BrowserStack people, the first one at the end of October. This problem disappears and then re-appears again after some time.

Yeah, i remember, what i fun read that is :-).

> We'll notice it the first time PATCH fails again

Would be kind of weird of the bug, if by the time they fix that issue we leave the testswarm. Anyways, fully agree on the matter.

File: test/unit/ajax.js [hunk 2]
```diff
@@ -1578,16 +1578,19 @@ module( "ajax", {
...
+	if ( location.search.indexOf( "swarmURL=" ) === -1 || document.documentMode < 9 ) {
```
Comment (markelog) [inline test/unit/ajax.js]: Hm, we probably need to start use `documentMode` everywhere where IE exclusion is needed.

**Pull request merged**